windows-observ-lib/panels.libsonnet

```diff
@@ -449,5 +449,327 @@ local utils = commonlib.utils;
                               targets=[t.networkInPacketsPerSec, t.networkOutPacketsPerSec]
                             )
                             + commonlib.panels.network.timeSeries.traffic.withNegateOutPackets(),
+
+      alertsPanel: g.panel.alertList.new(
+                     'Windows Active Directory alerts'
+                   )
+                   + g.panel.alertList.options.UnifiedAlertListOptions.withAlertInstanceLabelFilter('job=~"${job:regex}", instance=~"${instance:regex}"'),
```

Comment:
Could you try to use `queriesSelector` or at least this.config.instanceLabels, this.config.groupLabels to make sure that filter is changed accordingly?

One idea I have is to go to file variables.libsonnet and generate new field there called `queriesSelectorAdvancedSyntax` that will be just like `queriesSelector` but with ${<label>:regex} . Then we can use it here.

Reply:
Due to the way labelsToPromQLSelector works in commonlib,
<img width="1265" alt="image" src="https://github.com/grafana/jsonnet-libs/assets/38274348/02ecd981-da01-4e81-bcb2-7f000bff80f8">
doing it this way would cause the output to look like this
`job=~"integrations/windows_exporter",job:regex=~"$job:regex",instance:regex=~"$instance:regex"`
Since it reuses the label for the label and variable it will cause both parts to receive the regex modification that I add into the instance labels. Also the labelsToPromQLSelector function doesn't account for the `{}` encapsulation of the variable, eg: advanced grafana syntax should look like this: `'${job:regex}'`
Because of this i extended the commonlibs util function and added an advanced version of the labelsToPromQLSelector. This should take care of this, but I think it would be worth adding this function into commonlibs itself since alert panels are a regular occurence.